code/go/0chain.net/blobbercore/handler/object_operation_handler.go

```diff
@@ -581,15 +582,14 @@ func (fsh *StorageHandler) RenameObject(ctx context.Context, r *http.Request) (i
 		return nil, common.NewError("invalid_parameters", "Invalid connection id passed")
 	}
 
+	mutex := lock.GetMutex(AllocationConnections, connectionID) // It will be unlocked from WithConnection();
+	mutex.Lock()
```

Comment:
We shouldn't. 
It will create a problem if we unlock it here. 
The flow is like that, allocation.GetAllocationChanges() function returns connectionObj. It marks the connection as a new connection if the connectionId is not present in the AllocationChange table. 
This function `RenameObject`, for example, doesn't "actually" save the record in DB. The record is saved in WithConection() method. So if unlock the lock in the `RenameObject` function, then there might be the case that before the record is saved in the DB, the other operation runs (with old connectionId) arrive. This connection is again marked as a new connection!
And this was happening during my local testing. I discussed this with @lpoli about this issue, and we discussed this approach.